src/Features/CSharp/Portable/BraceCompletion/AbstractCurlyBraceOrBracketCompletionService.cs

```diff
@@ -164,9 +161,9 @@ static TextLine GetLineBetweenCurlys(int closingPosition, SourceText text)
                 return text.Lines[closingBraceLineNumber - 1];
             }
 
-            static LinePosition GetIndentedLinePosition(Document document, SourceText sourceText, int lineNumber, IndentationOptions options, CancellationToken cancellationToken)
+            static LinePosition GetIndentedLinePosition(ParsedDocument document, SourceText sourceText, int lineNumber, IndentationOptions options, CancellationToken cancellationToken)
             {
-                var indentationService = document.GetRequiredLanguageService<IIndentationService>();
+                var indentationService = document.LanguageServices.GetRequiredService<IIndentationService>();
```

Comment:
it would be cute to have ISynchronousLanguageService and force ParsedDocument.LanguageServices to only return those :)

Reply:
Good idea.